### PR TITLE
Adding tests of keyutils

### PR DESCRIPTION
**Pull request by nasastry**
security/keyutils-tests.py adds keyutils tests

Signed-off-by: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
Job log:

```
JOB ID     : 46e60fde3c54f3154e5fe67fc7c3835b6f4815d0
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-22T11.08-46e60fd/job.log
 (1/1) keyutils-tests.py:KeyUtils.test: FAIL: 5 test(s) failed, please refer to the log (194.32 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-22T11.08-46e60fd/results.html
JOB TIME   : 195.15 s
```

**Comment by nasastry** [comment]
Job log after fixing unused local variable:
```
JOB ID     : ef61be5a0bda845b9b7626de5a445de0efce6c0a
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-22T11.59-ef61be5/job.log
 (1/1) keyutils-tests.py:KeyUtils.test: FAIL: 5 test(s) failed, please refer to the log (193.96 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-22T11.59-ef61be5/results.html
JOB TIME   : 194.65 s
```

File: security/keyutils-tests.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        smm = SoftwareManager()
+        deps = ['gcc', 'make']
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+        url = "https://git.kernel.org/pub/scm/linux/kernel/git/dhowells/keyutils.git"
+
+        git.get_repo(url, destination_dir=self.logdir)
+        os.chdir(self.logdir + '/tests/')
```
Comment (harish-24) [inline security/keyutils-tests.py:45]: better to use `os.path.join()`

File: security/keyutils-tests.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        process.run("export AUTOMATED=1", shell=True)
```
Comment (harish-24) [inline security/keyutils-tests.py:46]: `os.environ()` would work?

File: security/keyutils-tests.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    def test(self):
+        '''
+        Running tests from keyutils
+        '''
+        count = 0
+        output = build.run_make(self.logdir + '/tests/',
```
Comment (harish-24) [inline security/keyutils-tests.py:53]: same here

File: security/keyutils-tests.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                                process_kwargs={"ignore_status": True})
+        for line in output.stdout_text.splitlines():
+            if 'FAILED' in line:
+                count += 1
+                self.log.info(line)
+        if count > 0:
```
Comment (harish-24) [inline security/keyutils-tests.py:59]: `if count:` would suffice

**Review by harish-24 — CHANGES_REQUESTED** [review]

**Comment by nasastry** [comment]
```
JOB ID     : 7095bae05eb85b82bf3713d97d85630e6056ee75
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-24T05.38-7095bae/job.log
 (1/1) keyutils-tests.py:KeyUtils.test: FAIL: 5 test(s) failed, please refer to the log (193.16 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-24T05.38-7095bae/results.html
JOB TIME   : 193.98 s
```

**Comment by nasastry** [comment]
@harish-24 review comments taken care. Please review. Thanks!!

**Comment by nasastry** [comment]
After all changes job log
```
JOB ID     : e785991f325fe336cbffa98b70a26e2c771a0e22
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-24T06.24-e785991/job.log
 (1/1) keyutils-tests.py:KeyUtils.test: FAIL: 5 test(s) failed, please refer to the log (197.45 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-24T06.24-e785991/results.html
JOB TIME   : 198.37 s
```

**Review by harish-24 — APPROVED** [review]
LGTM

**Pull request merged**